cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -546,7 +546,8 @@ protected SparkCollection<Object> handleJoin(Map<String, SparkCollection<Object>
       Map<String, Schema> inputSchemas = new HashMap<>();
       for (String inputStageName : pipelinePhase.getStageInputs(stageName)) {
         StageSpec inputStageSpec = pipelinePhase.getStage(inputStageName);
-        inputSchemas.put(inputStageName, inputStageSpec.getOutputSchema());
+        inputSchemas.put(inputStageName,
+                         inputStageSpec.getOutputPorts().values().stream().findFirst().orElse(null).getSchema());
```

Comment:
1. why the first port ?
2. what will be the case when it's null ? and when it's null, it will throw NPE ?